docs/devel/owners.md

```diff
@@ -0,0 +1,72 @@
+# Owners files
+
+_Note_: This is a design for a feature that is not yet implemented.
+
+## Overview
+We want to establish owners for different parts of the code in the Kubernetes codebase.  These owners
+will serve as the approvers for code to be submitted to these parts of the repository.  Notable, owners
+are not necessarily expected to do the first code review for all commits to these areas, but they are
+required to approve changes before they can be merged.
+
+## High Level flow
+
+### Step One: A PR is submitted
+After a PR is submitted, a user can request the approval from the required owners.  The user requests this
+by writing:
+```
+@k8s-bot approvers
+```
+
+into a comment.  When our automation sees this comment, it will reply with an annotation that indicates
+the owners required to approve.  The annotation is a comment that is applied to the PR.
+This comment will say:
+
+```
+Approval is required from <owner-a>, <owner-b>
+```
+
+PR authors are not required to ask for approval, if for example, the reviewer is already an owner, it is
+expected that the reviewer will (eventually) grant approval without being explicitly asked to do so.
+
+### Step Two: A PR is LGTM'd
+Once a PR is reviewed and LGTM'd it is eligible for submission.  However, in order for it to be submitted
+all owners have to 'approve' the PR.  Approval comes in two forms:
+
+   * An owner adds a comment to the PR saying "I approve"
+   * An owner is the original author of the PR
+
+In the case of a comment based approval, the same rules as for the 'lgtm' label apply.  If the PR is 
+changed by pushing new commits to the PR, the previous approval is invalidated, and the owner(s) must
+approve again.  Because of this is recommended that PR authors squash their PRs prior to getting approval
+from owners.
+
+### Step Three: A PR is merged
+Once a PR is LGTM'd and all owners have approved, it is eligible for merge.  The merge bot takes care of
+the actual merging.
+
+## Design details
+
+We need to build new features into the existing github munger in order to accomplish this.  Additionally
+we need to add owners files to the repository.
+
+### Approval Munger
+We need to add a munger that adds comments to PRs indicating whos approval they require.  This munger will
+look for PRs where approvers have been requested, and add an appropriate comment to the PR.
+
+
+### Status Munger
+GitHub has a [status api](https://developer.github.com/v3/repos/statuses/), we will add a status munger that pushes a status onto a PR of approval status.  This status will only be approved if the relevant
+approvers have approved the PR.
+
+### Requiring approval status
+Github has the ability to [require status checks prior to merging](https://help.github.com/articles/enabling-required-status-checks/)
+
+Once we have the status check munger described above implemented, we will add this required status check
+to our main branch as well as any release branches.
```

Comment:
How will we handle owners on release branches? Will we read the owners files from the branch or from master? Or have separate branch owners?

(I think that the bot doesn't do a good job today of distinguishing cherry picks from master merges)